### PR TITLE
test framework: use iop for integration test defaults

### PR DESCRIPTION
**Pull request by dgn**

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

**Comment by dgn** [comment]
I had previously tried to remove some of the e2e values files with this PR, but I'm now just moving them to `tests/e2e/test-values` for easier removal later on.

**Comment by dgn** [comment]
This now includes the changes from the reverted PR #21065. The postsubmit tests were broken because they needed the defaults from `test-vales/values-integ.yaml`. As this PR includes moving the integration test defaults to an IOP, they pass.

**Review by elfinhe — APPROVED** [review]

File: tests/istio.mk [hunk 1]
```diff
@@ -48,13 +48,13 @@ e2e_bookinfo_trustdomain: build test/local/auth/e2e_bookinfo_trustdomain
...
 	--egress=false --ingress=false \
-	--valueFile test-values/values-e2e.yaml \
+	--valueFile tests/e2e/values-e2e.yaml \
 	--rbac_enable=false --cluster_wide ${E2E_ARGS} ${T} ${EXTRA_E2E_ARGS}
 
 e2e_simple_noauth_run:
 	go test -v -timeout ${E2E_TIMEOUT} ./tests/e2e/tests/simple -args --auth_enable=false \
 	--egress=false --ingress=false \
-	--valueFile test-values/values-e2e.yaml \
+	--valueFile tests/e2e/values-e2e.yaml \
```
Comment (ericvn) [inline tests/istio.mk]: Does this need the `test-values` directory added? Same on line 51.

Comment (dgn) [reply tests/istio.mk]: You're right. In a previous iteration I only moved them into `tests/e2e`

Comment (dgn) [reply tests/istio.mk]: Done

**Comment by dgn** [comment]
This seems to need security and policies approvals, @yangminzhu @nrjpoddar could you take a look?

**Review by ostromart — APPROVED** [review]

**Review by elfinhe — APPROVED** [review]

**Review by myidpt — COMMENTED** [review]
/lgtm
/cc @lei-tang @JimmyCYJ

**Review by JimmyCYJ — APPROVED** [review]

**Review by douglas-reid — APPROVED** [review]
the telemetry changes are isolated and small and lgtm. approving for telemetry related changes.

**Comment by frankbu** [comment]
@dgn Deleting the helm charts in this PR is breaking the doc build:
```
 - ./public/docs/reference/config/installation-options/index.html
  *  External link https://github.com/istio/istio/tree/master/install/kubernetes/helm/istio failed: 404 No error
```
PRs that delete or reorganize structure in istio/istio should have a corresponding istio/istio.io PR to remove any broken references.

**Comment by dgn** [comment]
> @dgn Deleting the helm charts in this PR is breaking the doc build:
> 
> ```
>  - ./public/docs/reference/config/installation-options/index.html
>   *  External link https://github.com/istio/istio/tree/master/install/kubernetes/helm/istio failed: 404 No error
> ```
> 
> PRs that delete or reorganize structure in istio/istio should have a corresponding istio/istio.io PR to remove any broken references.

I will take care of it, thanks for the pointer. Note that I only deleted the last remnants of that directory (some old values files), the helm charts are long gone.

**Comment by dgn** [comment]
I see you already fixed it in https://github.com/istio/istio.io/pull/6961 -- thank you!

**Pull request merged**